File: SudokuSolver.cpp
// SudokuSolver.cpp
// Written by Jonathan Beamer
// Copyright (c) 2005-2008

#include <stdio.h>
#include "sudokuboard.h"

#define VERSION_STRING		"1.1"

int
main(int argc, char* argv[])
{
	int iterations;

	printf("Sudoku Solver v%s\nBy Jonathan Beamer\nCopyright (c) 2005-2016\n", VERSION_STRING);
	printf("Input file: %s\n", (argc>1 ? argv[1] : "none"));

	// TODO -- board size should be set by the puzzle file read in
	CSudokuBoard cBoard("1", 9);
	if (!cBoard.ReadInFile( (argc>1 ? argv[1] : NULL) ))
		return 1;

	printf("\nUnsolved Puzzle:\n");
	cBoard.Print();

	iterations = cBoard.Solve();

	printf("\nSummary:\n  Solutions:  %d\n  Branches:   %d\n  Iterations: %d\n\n", g_nSolutionsFound, g_nBranchEndpoitsFound, iterations);
	return 0;
}

File: pyPort/SudokuSolver.py
# SudokuSolver.py
# Written by Jonathan Beamer
# Copyright (c) 2005-2016
# --
# python port September, 2016

import sys
import SudokuBoard

# global variables
VERSION_STRING = "1.3"
COPYRIGHT_STRING = "Copyright (c) 2005-2021"
USAGE = "Usage: python SudokuSolver.py [puzzle_filename]"


def main():
    print(f'Sudoku Solver v{VERSION_STRING}\nBy Jonathan Beamer\n{COPYRIGHT_STRING}')

    if len(sys.argv) < 2:
        print(USAGE)
        sys.exit(1)

    board = SudokuBoard.SudokuBoard("1")
    if (board.ReadInFile(sys.argv[1])):
        print(USAGE)
        sys.exit(1)

    print('\nUnsolved Puzzle:')
    board.PrintCompact()

    nSolutionsFound = board.Solve()


if __name__ == "__main__":
    main()
